addons/calendar/models/calendar_recurrence.py

```diff
@@ -361,6 +362,14 @@ def _rrule_parse(self, rule_str, date_start):
         data = {}
         day_list = ['mon', 'tue', 'wed', 'thu', 'fri', 'sat', 'sun']
 
+        # Skip X-named RRULE extensions
+        # TODO Remove patch when dateutils contains the fix
+        # HACK https://github.com/dateutil/dateutil/pull/1374
+        # Optional parameters starts with X- and they can be placed anywhere in the RRULE string.
+        # RRULE:FREQ=MONTHLY;INTERVAL=3;X-RELATIVE=1
+        # RRULE;X-EVOLUTION-ENDDATE=20200120:FREQ=WEEKLY;COUNT=3;BYDAY=MO
+        rule_str = re.sub(r';?X-[-\w]+=[^;:]*', '', rule_str)
```

Comment:
The regexp is simplified so it supports lowercase param names (RFC doesn't specify they *have* to be uppercase AFAICS). It also removes the optionally leading semicolon, so all is done in a single step.

Reply:
Thanks, it is way better than mine indeed. I still have an issue with this case:
```
RRULE:X-FOO=BAR;FREQ=WEEKLY;COUNT=3;BYDAY=MO;X-RELATIVE=1
```
In that case, rule_str becomes 
```
RRULE:;FREQ=WEEKLY;COUNT=3;BYDAY=MO;X-RELATIVE=1
```
Not sure if we should manually replace by `RRULE:FREQ...` in a second step.

I created a small playground with the case I thout about based on your tests: https://regex101.com/r/R9gonm/1

Reply:
Good catch! I hardened the test and fixed that case.